### PR TITLE
feat(api): moved virtualization api structs to separated go mod

### PR DESCRIPTION
**Pull request by Isteb4k**
## Description

Moved virtualization api to separated go module 

## Why do we need it, and what problem does it solve?
It is necessary to be able to import only api entities into other projects without the virtualization controller internals

## What is the expected result?
New go module 

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

### CODE REVIEW

File: api/scripts/update-codegen.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+function usage {
+  cat <<EOF
+Usage: $(basename "$0") <core/subresources/all>
+Example:
+   $(basename "$0") controller
```
Comment (yaroslavborbat) [inline api/scripts/update-codegen.sh:10]:
```suggestion
   $(basename "$0") core
```

**Review by yaroslavborbat — APPROVED** [review]

File: images/virtualization-artifact/Taskfile.dist.yaml [hunk 2]
```diff
@@ -682,6 +658,22 @@ tasks:
...
+    cmd: task api:generate:core
+
+  generate:subresources:
+    desc: "Regenerate code for subresources components"
+    cmd: task api:generate:subresources
+
+  ci:generate:
+    desc: "Run generations and check git diff to ensure all files are committed"
+    cmd: task api:ci:generate
+
```
Comment (diafour) [inline images/virtualization-artifact/Taskfile.dist.yaml:676]: I think these shortcuts are not needed. task list will show all api:NNNNN tasks.

File: images/virtualization-artifact/go.mod [hunk 3]
```diff
@@ -3,20 +3,17 @@ module github.com/deckhouse/virtualization-controller
 go 1.21
 
 require (
+	github.com/deckhouse/virtualization/api v0.0.0-20240321085500-118bb9c256f6
```
Comment (diafour) [inline images/virtualization-artifact/go.mod:6]: Oops, `replace` is gone.

**Review by diafour — APPROVED** [review]
Just add replace.

**Pull request merged**